Feature/calendar links and events

doingandlearning opened the pull request:
This PR adds two features:
- A Google calendar link
- A dynamic ICS file generation and download

To test:
- Navigate to an event
- Click on the Google Calendar event
- Confirm that the details match
- Click on the download ICS button
- Use a tool such as this one https://icalendar.org/validator.html#results to check that it is valid

doingandlearning committed:
feature: add generateCalendarScript

doingandlearning committed:
feature: adds calendar to session page.

doingandlearning committed:
Adds two new libraries

doingandlearning committed:
Adds new calendar events.

doingandlearning committed:
Merge branch 'master' into feature/calendar-links-and-events

doingandlearning committed:
Updates to script.

doingandlearning committed:
Tidies the code for calendar.

doingandlearning committed:
Updates the Google calendar link.

doingandlearning committed:
Merge branch 'master' into feature/calendar-links-and-events

doingandlearning committed:
Updates script with new meta data.

doingandlearning committed:
Adds google link

doingandlearning committed:
Tidies up styling.

doingandlearning committed:
Merge branch 'master' into feature/calendar-links-and-events

doingandlearning committed:
fix: removes unnecessary method.

doingandlearning committed:
fix Remove merge conflict artifact.

doingandlearning committed:
fix Remove duplicate

doingandlearning committed:
Adds default location to communities object.

doingandlearning committed:
fix Adds fallback string.

phivk committed:
Merge branch 'master' into feature/calendar-links-and-events

phivk committed:
Merge branch 'master' into feature/calendar-links-and-events

# Conflicts:
#	content/mozfest/sessions/template.markdown

doingandlearning committed:
Merge branch 'master' into feature/calendar-links-and-events

phivk committed:
add Barts & Drafts test event

NB: remove before merging!

doingandlearning committed:
Adds dynamic generation of ICS file.

doingandlearning committed:
Removes the ICS library.

doingandlearning committed:
Remove the static ICS files.

doingandlearning committed:
Merge branch 'develop' into feature/calendar-links-and-events

doingandlearning committed:
Removes legacy ics provision.

doingandlearning committed:
Fixes hasHappend function.

doingandlearning committed:
Adds a br

phivk reviewed (changes requested):
I havent completed a full review yet, but wanted to share this first piece of feedback.
navigating to session pages from the home page works fine, but when I navigate directly from the browser to 
http://localhost:3000/sessions/current/demo-upcoming
I get this error:
```
 this.session.title.toLowerCase(...).replaceAll is not a function 
```

![image](https://user-images.githubusercontent.com/902958/181825803-24e0574f-610f-495d-bc3b-396ad2c36293.png)

phivk committed:
ensure upcoming session date is actually upcoming

phivk commented:
@doingandlearning 
I tried opening the downloaded .ics file in the Apple Calendar app and got this error:
<img width="265" alt="481E0CCB-C9E8-402F-814A-5F542A9C0C21-1303-00008F4F3DA7A844" src="https://user-images.githubusercontent.com/902958/182040869-3cbb136f-6b87-4b45-a2e2-be8bd5c46bf3.png">

will share the respective ics file via slack

phivk commented:
> I havent completed a full review yet, but wanted to share this first piece of feedback. navigating to session pages from the home page works fine, but when I navigate directly from the browser to http://localhost:3000/sessions/current/demo-upcoming I get this error:
> 
> ```
>  this.session.title.toLowerCase(...).replaceAll is not a function 
> ```
> 
> ![image](https://user-images.githubusercontent.com/902958/181825803-24e0574f-610f-495d-bc3b-396ad2c36293.png)

This was due to the fact I was running an older version (12) of node locally. It works fine when I use v16 like we are using on Vercel for deployments.

phivk commented:
> @doingandlearning I tried opening the downloaded .ics file in the Apple Calendar app and got this error: <img alt="481E0CCB-C9E8-402F-814A-5F542A9C0C21-1303-00008F4F3DA7A844" width="265" src="https://user-images.githubusercontent.com/902958/182040869-3cbb136f-6b87-4b45-a2e2-be8bd5c46bf3.png">
> 
> will share the respective ics file via slack

I found these two errors via https://icalendar.org/validator.html:

1. Lines not delimited by CRLF sequence [near line # 1](javascript:void(0);)Reference: [RFC 5545 3.1. Content Lines](http://icalendar.org/iCalendar-RFC-5545/3-1-content-lines.html)
2. Invalid DTSTAMP value, must be a valid date-time value [near line # 7](javascript:void(0);)Reference: [3.3.5. Date-Time](http://icalendar.org/iCalendar-RFC-5545/3-3-5-date-time.html)

phivk committed:
fix Invalid DTSTAMP value

from .ics validator (https://icalendar.org/validator.html):
Invalid DTSTAMP value, must be a valid date-time value near line # 7Reference: 3.3.5. Date-Time

phivk committed:
fix Lines not delimited by CRLF sequence

from .ics validator (https://icalendar.org/validator.html):
Lines not delimited by CRLF sequence near line # 1Reference: RFC 5545 3.1. Content Lines

phivk committed:
refactor ICS and GoogleCalendar generation into util/date.js

phivk committed:
refactor ICS date generation to use dayjs for UTC formatting
